ci/test_suites/authenticators_gcp/get_gcp_id_tokens.sh

```diff
@@ -122,18 +127,20 @@ get_tokens_into_files() {
     error_exit "-- Cannot run command, GCE instance '${INSTANCE_NAME}' not in a valid state!"
   fi
 
-  local tokens_config_file="$(dirname $0)/tokens_config.json"
+  local tokens_config_file
+  tokens_config_file="$(dirname "$0")/tokens_config.json"
   if [ ! -f "$tokens_config_file" ]; then
     echo "$tokens_config_file file not found."
     exit 1
   fi
 
-  local tokens="$(cat $tokens_config_file)"
+  local tokens
+  tokens=$(< "$tokens_config_file")
   local token_prefix="${TOKENS_OUT_DIR_PATH}/${TOKEN_FILE_NAME_PREFIX}"
 
   for row in $(echo "${tokens}" | jq -c '.[]'); do
     _jq() {
-      echo ${row} | jq -r ${1}
+      echo "$row" | jq -r "$1"
```

Comment:
Ive seen both `${}` and just `$`. Are they syntactically the same? Can we choose one align throughout the repo?

Reply:
@sigalsax You should use `$1` or `$var` when you can, and `${1}` or `${var}` when you have to.  In some contexts they are the same, in others not, so you can't unilaterally say to always use one.  Rather, you _could_ decide to always use `${var}` but you'll add a lot of visual noise for no reason to the 90% case.  Better to understand what they actual do.

When you are using arrays or parameter expansions, the `${}` form is required.  It's also required if you want to interpolate your variable and have text directly abutting it: 

```sh
$ sigal=avocado
$ echo "$sigals RULE"
 RULE
$ echo "${sigal}s RULE"
avocados RULE
```
There are probably some other cases where you need `${}` I'm forgetting too.  But most of the time, for simple cases, the plain form `$var` works fine and should be preferred.

Reply:
Gotcha and might I say, love that I am equated to an 🥑